tests/cross/src/test/scala/tests/pc/CompletionScalaCliSuite.scala

```diff
@@ -105,6 +105,17 @@ class CompletionScalaCliSuite extends BaseCompletionSuite {
     "better-tostring",
   )
 
+  checkEdit(
+    "alternative-sorting",
+    """|//> using lib "co.fs2::fs2-core:@@"
```

Comment:
Out of curiosity, what was the format that was breaking sorting? Can we handle that in Version.fromString ?

Reply:
Side note: With some libraries like fs2 or cats-effect (can't remember for sure), scala-cli was suggesting an upgrade to sth like `3.3-commithashhere` when `3.3.4` was the latest release. Could it be that sort of format?

(I'll report the issue in scala-cli when I see it again)

Reply:
Yes, it was caused by format like `3.5-ac5c657`, but there are also versions like `3.2.14-75-7902cbf`. 
In `Version` we do 
```scala
val parts = version.split("\\.|-")
val Array(major, minor, patch) = parts.take(3).map(_.toInt)
```
Which fails trying to do `toInt` on commithash, so maybe we can handle it there somehow

Reply:
Potentially, we potentially drop any hash we encounter, since we can't get any semantic meaning from it.